### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -82,7 +91,12 @@ exports.register = (server, pluginOptions, next) => {
     routeOptions.handler = options.routeHandler ? options.routeHandler : (request, reply) => {
       // the report will be a Buffer representing a JSON string:
       if (request.payload) {
-        server.log(options.logTags, request.payload.toString('utf-8'));
+        try {
```
Comment (jgallen23) [inline index.js]: lets do a try catch around JSON.parse and if it fails, just do a payload.toString and then send either of those to server.log

File: index.js [hunk 2]
```diff
@@ -32,24 +33,32 @@ exports.register = (server, pluginOptions, next) => {
...
-    if (options.varietiesToInclude.indexOf(request.response.variety) < 0) {
-      return reply.continue();
+    // unless the cspHeader option is set for this route, check if we still need to intercept the response:
+    if (!request.route.settings.plugins['hapi-csp-mixed'] || !request.route.settings.plugins['hapi-csp-mixed'].cspHeaders) {
+      // don't worry about it if we are only doing https routes and this isn't https:
+      if (options.httpsOnly && request.server.info.protocol !== 'https') {
+        return reply.continue();
+      }
+      // don't worry about it if this was called by the CSP report route:
+      if (options.fetchDirectives['report-uri'] === request.path) {
```
Comment (jgallen23) [inline index.js]: I think this should be the first check

File: index.js [hunk 3]
```diff
@@ -32,24 +33,32 @@ exports.register = (server, pluginOptions, next) => {
...
   server.ext('onPreResponse', (request, reply) => {
-    // don't worry about it if this was called by the CSP report route:
-    if (options.fetchDirectives['report-uri'] === request.path) {
-      return reply.continue();
-    }
-    // don't worry about it if this response variety isn't in the indicated varieties:
-    if (options.varietiesToInclude.indexOf(request.response.variety) < 0) {
-      return reply.continue();
+    // unless the cspHeader option is set for this route, check if we still need to intercept the response:
+    if (!request.route.settings.plugins['hapi-csp-mixed'] || !request.route.settings.plugins['hapi-csp-mixed'].cspHeaders) {
```
Comment (jgallen23) [inline index.js]: should check plugin cspHeaders != true or varieties !include and then return reply.contine().  This should be the second check.

File: index.js [hunk 4]
```diff
@@ -32,24 +33,32 @@ exports.register = (server, pluginOptions, next) => {
...
+      // don't worry about it if we are only doing https routes and this isn't https:
+      if (options.httpsOnly && request.server.info.protocol !== 'https') {
```
Comment (jgallen23) [inline index.js]: then this is the 3rd check

File: index.js [hunk 5]
```diff
@@ -2,7 +2,8 @@
 const aug = require('aug');
 const url = require('url');
 const pluginDefaults = {
-  logTags: ['content-security-policy-report'],
+  httpsOnly: false, // only respond if request is https
```
Comment (jgallen23) [inline index.js:5]: is there any way to add a test for this?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js (hunk 5 above)
Comment (orthagonal) [reply index.js:5]: Added a test for this by just manually setting server.info.protocol before server.inject.

**Pull request merged**